src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,492 @@
+        } else {
+          "CRD"
+        },
+        licenceStatus = licence.licenceStatus,
+        nomisLegalStatus = p.prisoner.legalStatus,
+        lastWorkedOnBy = licence.updatedByFullName,
+        isDueForEarlyRelease = licence.isDueForEarlyRelease,
+        isInHardStopPeriod = licence.isInHardStopPeriod,
+        tabType = determineCaViewCasesTab(
+          p.prisoner,
+          p.cvl,
+          licence,
+        ),
+        probationPractitioner = ProbationPractitioner(staffUsername = licence.comUsername),
+      )
+    }.filterNotNull()
+  }
+
+  private fun determineCaViewCasesTab(
+    nomisRecord: Prisoner,
+    cvlFields: CvlFields,
+    licence: LicenceSummary?,
+  ): CaViewCasesTab {
+    val releaseDate = (licence?.actualReleaseDate ?: licence?.conditionalReleaseDate) ?: selectReleaseDate(
+      nomisRecord,
+    )
+
+    if (licence != null &&
+      isAttentionNeeded(
+        licence.licenceStatus,
+        licence.licenceStartDate,
+        releaseDate,
+      )
+    ) {
+      return CaViewCasesTab.ATTENTION_NEEDED
+    }
+    val isDueToBeReleasedInTheNextTwoWorkingDays =
+      licence?.isDueToBeReleasedInTheNextTwoWorkingDays ?: cvlFields.isDueToBeReleasedInTheNextTwoWorkingDays
+
+    return when {
+      isDueToBeReleasedInTheNextTwoWorkingDays -> CaViewCasesTab.RELEASES_IN_NEXT_TWO_WORKING_DAYS
+      else -> CaViewCasesTab.FUTURE_RELEASES
+    }
+  }
+
+  private fun isAttentionNeeded(
+    status: LicenceStatus,
+    licenceStartDate: LocalDate?,
+    releaseDate: LocalDate?,
+    overrideClock: Clock? = null,
+  ): Boolean {
+    val now = overrideClock ?: clock
+    val today = LocalDate.now(now)
+
+    val noReleaseDates = releaseDate == null
+
+    val missingDates = listOf(
+      IN_PROGRESS,
+      SUBMITTED,
+      APPROVED,
+      NOT_STARTED,
+    ).contains(status) &&
+      noReleaseDates
+    val startDateInPast = licenceStartDate != null && status == APPROVED && licenceStartDate.isBefore(today)
+
+    return missingDates || startDateInPast
+  }
+
+  private fun selectReleaseDate(nomisRecord: Prisoner): LocalDate? {
+    return nomisRecord.confirmedReleaseDate ?: nomisRecord.conditionalReleaseDate
+  }
+
+  private fun getPrisonersApproachingRelease(
+    prisonCaseload: List<String>?,
+    overrideClock: Clock? = null,
+  ): Page<CaseloadItem>? {
+    val now = overrideClock ?: clock
+    val weeksToAdd: Long = 4
+    val today = LocalDate.now(now)
+    val todayPlusFourWeeks = LocalDate.now(now).plusWeeks(weeksToAdd)
+    return prisonCaseload?.toSet()?.let {
+      caseloadService.getPrisonersByReleaseDate(
+        today,
+        todayPlusFourWeeks,
+        it,
+        page = 0,
+      )
+    }
+  }
+
+  fun findLatestLicenceSummary(licences: List<LicenceSummary>): LicenceSummary? {
+    if (licences.size == 1) {
+      return licences[0]
+    }
+    if (licences.any { it.licenceStatus == TIMED_OUT }) {
+      return licences.find { it.licenceStatus != TIMED_OUT }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏This will return the first but do we need to check it is the most recent worked on licence if there are multiple licences or in order of preference of the licence status?

Reply:
Ya, will add sort by while fetching licences this should solve the order.
**sortBy = "conditionalReleaseDate"**

Reply:
There should never be more than two licences if one of them is `TIMED_OUT`
One would be the NOT_STARTED/IN_PROGRESS licence that was timed out, and the optional other is the prison-created licence, if they've started it. This means that if there's a TIMED_OUT licence and more than one licence, we always want to return the other one